src/targets/awsLambda.ts

```
@@ -0,0 +1,139 @@
+  }
+
+  /**
+   * Extracts AWS Lambda target options from the environment.
+   */
+  protected getAwsLambdaConfig(): AwsLambdaTargetOptions {
+    if (!process.env.AWS_ACCESS_KEY_ID || !process.env.AWS_SECRET_ACCESS_KEY) {
+      throw new ConfigurationError(
+        `Cannot perform AWS Lambda release: missing credentials.
+        Please use AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables.`
+      )
+    }
+    return {
+      awsAccessKeyId: process.env.AWS_ACCESS_KEY_ID,
+      awsSecretAccessKey: process.env.AWS_SECRET_ACCESS_KEY,
+    }
+  }
+
+  /**
+   * Publishes current lambda layer zip bundle to AWS Lambda.
+   * @param _version New version to be released.
+   * @param revision Git commit SHA to be published.
+   */
+  public async publish(_version: string, revision: string): Promise<any> {
+    logger.debug('Fetching artifact list...');
+    const packageFiles = await this.getArtifactsForRevision(revision, {
+      includeNames: DEFAULT_AWS_LAMBDA_DIST_REGEX,
+    })
+
+    if (packageFiles.length == 0) {
+      reportError('Cannot release to AWS Lambda: no packages found');
+      return undefined;
+    } else if(packageFiles.length > 1) {
+      reportError(`Cannot release to AWS Lambda:
+        multiple packages with matching patterns were found.`);
+      return undefined;
+    }
+
+    const artifactBuffer = fs.readFileSync(
+      await this.artifactProvider.downloadArtifact(packageFiles[0])
+    );
+
+    for (let i = 0; i < awsAllRegions.length; i++) {
+      const currentRegion = awsAllRegions[i];
+      const lambda = new Lambda({ region: currentRegion });
+      const publishedLayer = await lambda
+        .publishLayerVersion({
+          Content: {
+            ZipFile: artifactBuffer,
+          },
+          LayerName: layerName,
+          CompatibleRuntimes: compatibleRuntimes,
+          LicenseInfo: 'MIT',
+        })
+        .promise();
+
+      if (publishedLayer.Version === undefined) {
```

@iker-barriocanal when would the `Version` be undefined?

---

As discussed, we should throw an error if `Version` is `undefined`.